misk/src/main/kotlin/misk/web/WebActionModule.kt

```diff
@@ -67,14 +72,17 @@ internal class BoundActionProvider<A : WebAction, R>(
     val httpMethod: HttpMethod
 ) : Provider<BoundAction<A, *>> {
 
-    @Inject lateinit var interceptorFactories: List<Interceptor.Factory>
+    @Inject lateinit var userProvidedInterceptorFactories: List<Interceptor.Factory>
+    @Inject lateinit var defaultInterceptorFactories: @MiskDefault List<Interceptor.Factory>
     @Inject lateinit var parameterExtractorFactories: List<ParameterExtractor.Factory>
 
     override fun get(): BoundAction<A, *> {
         val action = function.asAction()
 
         val interceptors = ArrayList<Interceptor>()
-        interceptorFactories.mapNotNullTo(interceptors) { it.create(action) }
+        // Ensure that default interceptors are called before any user provided interceptors
+        defaultInterceptorFactories.mapNotNullTo(interceptors) { it.create(action) }
```

Comment:
I find `mapNotNullTo` to be a very awkward method

Reply:
Yeah, it doesn't really map well coming from other languages. I'll see if I can make it more grok-able.